machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,154 @@
+// +k8s:openapi-gen=true
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// userDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// credentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// More detail about Power VS service instance
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+
+	// serviceInstance is the reference to the Power VS service on which the server instance(VM) will be created.
+	// Power VS service is a container for all Power VS instances at a specific geographic region.
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// serviceInstance identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// image is to identify the rhcos image uploaded to IBM COS bucket which is used to create the instance.
+	// image identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// network is the reference to the Network to use for this instance.
+	// network identifier (name or id) can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// keyPairName is the name of the KeyPair to use for SSH
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
+
+	// systemType is the System type used to host the instance.
+	// systemType determines the number of cores and memory that is available.
+	// Few of the supported SystemTypes are s922,e880,e980
+	// e880 systemType available only in Dallas Datacenters
+	// e980 systemType available in Datacenters except Dallas and Washington
+	// Defaults to s922 which is generally available.
+	// +optional
+	SystemType string `json:"systemType,omitempty"`
+
+	// processorType is the VM instance processor type.
+	// It must be set to one of the following values: dedicated, capped or shared.
+	// dedicated: resources are allocated for a specific client, The hypervisor makes a 1:1 binding of a partition’s processor to a physical processor core.
+	// shared: shared among other clients
+	// capped: shared, but resources do not expand beyond those that are requested, the amount of CPU time is capped to the value specified for the entitlement.
+	// if the processorType is selected as dedicated, then processors value cannot be fractional
+	// Defaults to shared.
+	// +kubebuilder:validation:Enum:="dedicated";"shared";"capped"
```

Comment:
These values should use pascal case as this is the convention for Kubernetes like resources
```suggestion
	// +kubebuilder:validation:Enum:="Dedicated";"Shared";"Capped"
```

Reply:
There are instances that this is not followed ref: https://cs.k8s.io/?q=kubebuilder%3Avalidation%3AEnum&i=nope&files=&excludeFiles=&repos= and also we need to convert to small case in the controller.